### ISSUE TITLE
Results Inconsistent with the Original Paper on Three Datasets

### ISSUE THREAD

**Issue by liyu008** [2220210616YR/TCompoundE#4]
We have executed the code you provided; however, **we observed that the actual results on three datasets significantly deviate from the experimental outcomes reported in your original paper**. Herein, we present a comparison between the actual results and those reported in the paper to illustrate this discrepancy.

### ICEWS14
For the **ICEWS14** dataset, the results in your paper are:
<img width="393" alt="截屏2024-12-14 18 27 25" src="https://github.com/user-attachments/assets/b8040838-5c6d-4525-9d53-a37d49c6e46f" />
We proceeded to execute the following instructions in accordance with your requirements:
`python learner.py --dataset ICEWS14 --emb_reg 0.01 --time_reg 0.01 --learning_rate 0.01  --rank 6000  --batch_size 4000  --max_epochs 400`
**Here is the actual results:** 
<img width="766" alt="截屏2024-12-14 18 20 47" src="https://github.com/user-attachments/assets/c059f20d-050f-43b0-9ad2-2c20afa6252f" />

### ICEWS05-15
In terms of **ICEWS05-15** dataset, the results reported in your paper are:
<img width="263" alt="截屏2024-12-14 18 37 49" src="https://github.com/user-attachments/assets/eb693431-4a8e-4941-8f3f-482bfce9e7c2" />
We proceeded to execute the following instructions in accordance with your requirements:
`python learner.py --dataset ICEWS05-15 --emb_reg 0.002   --time_reg 0.1  --learning_rate 0.08  --rank 8000  --batch_size 6000  --max_epochs 100`
**Here is the actual results:** 
<img width="761" alt="截屏2024-12-14 18 42 53" src="https://github.com/user-attachments/assets/369167d8-5c32-4bad-9e35-466946b3aa69" />

### GDELT
Regarding GDELT dataset, the results presented in your paper are:
<img width="255" alt="截屏2024-12-14 18 51 55" src="https://github.com/user-attachments/assets/7214b02a-9b69-43c0-b395-cab5fbcdb875" />
We proceeded to execute the following instructions in accordance with your requirements:
`python learner.py --dataset GDELT --emb_reg 0.001   --time_reg 0.001  --learning_rate 0.35  --rank 6000  --batch_size 2000  --max_epochs 50`
**Here is the actual results:** 
<img width="772" alt="截屏2024-12-14 19 11 04" src="https://github.com/user-attachments/assets/1796ae01-4eb8-46a7-ac3a-86fa0ad9e26c" />

### Further Questions
Additionally, we have a few other questions regarding the code you provided:

1. We attempted to validate whether the suboptimal results were due to the hyper-parameters (e.g., lr, batch size, emb_reg, time_reg, ...) by modifying those provided. However, we found that regardless of how we adjusted the hyper-parameters, the performance remained consistently subpar, with no significant variation in the outcomes.
2. During the training process, we observed that the model's performance on the training set was notably poor in each epoch, even worse than its performance on the validation set. This observation leads us to question whether there might be an inherent flaw in the architecture of the model itself.
Here is an example:
<img width="814" alt="截屏2024-12-14 19 14 32" src="https://github.com/user-attachments/assets/b9360936-5ffb-4c8b-80bb-a213c8900392" />

Thus, we sincerely hope you can address these issues to confirm the validity and consistency of your work with the conclusions in the paper.
